src/pykeen/trackers.py

```diff
@@ -39,14 +39,36 @@ def end_run(self) -> None:
 class MLFlowResultTracker(ResultTracker):
     """A tracker for MLFlow."""
 
-    def __init__(self, tracking_uri: Optional[str] = None):
+    def __init__(
+        self,
+        tracking_uri: Optional[str] = None,
+        experiment_id: Optional[int] = None,
+        experiment_name: Optional[str] = None,
+    ):
+        """
+        Initialize result tracking via MLFlow.
+
+        :param tracking_uri:
+            The tracking uri.
+        :param experiment_id:
+            The experiment ID. If given, this has to be the ID of an existing experiment in MFLow. Has priority over
+            experiment_name.
+        :param experiment_name:
+            The experiment name. If this experiment name exists, add the current run to this experiment. Otherwise
+            create an experiment of the given name.
+        """
         import mlflow as _mlflow
         self.mlflow = _mlflow
 
-        if tracking_uri is None:
-            tracking_uri = 'localhost:5000'
-
         self.mlflow.set_tracking_uri(tracking_uri)
+        if experiment_id is not None:
+            experiment = self.mlflow.get_experiment(experiment_id=experiment_id)
+            self.mlflow.set_experiment(experiment.name)
+        elif experiment_name is not None:
+            experiment = self.mlflow.get_experiment_by_name(name=experiment_name)
+            if experiment is None:
+                experiment = self.mlflow.create_experiment(name=experiment_name)
```

Comment:
with MLflow 1.8+, the experiment id is returned as a string. It appears you might just be able to get away with using set_experiment() on a name and not have to go through this middle step

Reply:
Addressed in https://github.com/pykeen/pykeen/pull/40/commits/a9568105718cc5b78eda9837fae33c5565e0e24d